src/fasta.rs

```diff
@@ -0,0 +1,210 @@
+// This module is for reading and writing FASTA format files
+
+use crate::DnaSequence;
+use std::io::BufRead;
+use std::str::FromStr;
+
+pub struct SimpleFastaParser {}
+
+impl SimpleFastaParser {
+    pub fn parse<R: BufRead>(handle: &mut R) -> Result<Vec<(String, String)>, anyhow::Error> {
+        let mut result: Vec<(String, String)> = vec![];
+
+        let mut lines: Vec<String> = vec![];
+        let mut title: Option<String> = None;
+
+        for line in handle.lines().flatten() {
```

Comment:
flattening here will ignore file read errors instead of propagating them

Reply:
yeah. So does the python code.

Reply:
Yeah, I think so. We could maybe have a variant that swallows errors, but I wouldn't want that to be the default behavior since it could lead to silent data loss (e.g., in our pipeline we don't want parts of a virus to silently be not read and not included).

```suggestion
        for line in handle.lines() {
            let line = line?;
```

and expose the `io::Error` or whatever.